code/modules/reagents/chemistry/reagents/drink_reagents.dm

```diff
@@ -209,6 +210,31 @@
 		holder.remove_reagent(/datum/reagent/consumable/capsaicin, 2)
 	..()
 
+/*See block comment in ../milk/overdose_process(mob/living/M) for calculation and explanation of why this exists and why 5 was chosen
+* For best results use in tandem with method outlined in this comment
+*/
+/datum/reagent/consumable/milk/overdose_start(mob/living/M)
+	M.reagents.add_reagent(/datum/reagent/toxin/bonehurtingjuice, 5) //The integer here should match var/starting_amount in ../milk/overdose_process(mob/living/M)
+	return ..()
+
+/datum/reagent/consumable/milk/overdose_process(mob/living/M)
+	var/datum/reagent/converted_reagent = /datum/reagent/toxin/bonehurtingjuice //Needed to get the metabolism for desired reagent, exists solely for brevity compared to /datum/reagent/category/reagent.metabolization_rate
+	var/minimum_cycles = overdose_threshold/metabolization_rate //minimum_cycles is the number of ticks for an amount of units equal to the overdose threshold to process.
+	var/amount_to_add = 45 / minimum_cycles + converted_reagent.metabolization_rate //amount_to_add is the calculated amount to add per tick to meet ensure that target_units after minimum_cycle ticks.
```

Comment:
```suggestion
	var/amount_to_add = 45 / minimum_cycles + initial(converted_reagent.metabolization_rate) //amount_to_add is the calculated amount to add per tick to meet ensure that target_units after minimum_cycle ticks.
```

Reply:
I still don't fully understand why you're grabbing the _reagent's_ metab rate instead of the mob's (which is the thing that matters). But this'll runtime without initial() because the datum `converted_reagent` is set to is a typepath

Reply:
If I you're referring to the stuff regarding the liver, I'm keeping it to the standard liver, as upgrades are meant to be upgrades. Otherwise I wasn't aware mobs can metabolize at different rates than the chems itself.

Reply:
iirc the reagent's metab var doesn't change. It's just used in liver calculations

Reply:
Well I guess I can take a look into liver code and see what goes on, and make an evaluation of whether or not accounting for that will cause any weirdness.

Reply:
Reagent metabolism is unaccounted for in liver code. Reagent amount is the only thing really cared about.